app/src/components/ChatBot/chat-bot.tsx

```diff
@@ -93,6 +332,62 @@ export default function ChatBot({
     },
   ];
 
+  /////////////////////
+  // Utility Helpers //
+  /////////////////////
+
+  const appendMessage = (role: Message["role"], text: string) => {
+    setMessages((prevMessages) => [...prevMessages, { role, text }]);
+  };
+
+  const appendToLastMessage = (text: string) => {
+    setMessages((prevMessages) => {
+      const lastMessage = prevMessages[prevMessages.length - 1];
+      const updatedLastMessage = {
+        ...lastMessage,
+        text: lastMessage.text + text,
+      };
+      return [...prevMessages.slice(0, -1), updatedLastMessage];
+    });
+  };
+
+  // TODO: Fix annotations
```

Comment:
is this TODO part of another ticket? If so we can add the ticket code.

Reply:
Done